### PR TITLE
Migrate to @emotion/core

### PR DESCRIPTION
**Pull request by liywjl**
## What does this change?
Update the use of `@emotion/core` to be more in line with how you write emotion 10.

## Why?
We have been using Emotion 10 but we write emotion code as if we are using Emotion 9 (this was allowed to help make the transition simple). But now as we are aiming to go to Emotion 11, we should first migrate to using Emotion 10 syntax.

## JSX?
Emotion uses `jsx` exposed from `@emotion/core` (in Emotion 10), rather than React, as a way of converting `css` props on components to `classNames`. In order to get this working you need to add a babel preset `@emotion/babel-preset-css-prop` and make sure that react files start with `/** @jsx jsx */` (a way to indicate to babel that you want to use JSX rather than React.

## Fragments?
Babel does not understand `<></>`, therefore we use React to explicitly define `React.Fragment`

## JSX Runtime?
We can only support JSX runtime classic untile we migrate to Emotion 11, so each file needs to have the pragma:`/** @jsxRuntime classic */` at the top

### CODE REVIEW

**Review by shtukas — APPROVED** [review]

File: babel.config.js [hunk 1]
```diff
@@ -1,10 +1,19 @@
...
+                "cssPropOptimization": true
+            }
+        ]
+    ],
     presets: [
         '@babel/preset-env',
         '@babel/preset-typescript',
-        '@babel/preset-react',
-        '@emotion/babel-preset-css-prop',
+        "@babel/preset-react",
```
Comment (SiAdcock) [inline babel.config.js:15]: lol them quotes

**Comment by buck06191** [comment]
Does this not work with regards to Babel and Fragments: https://babeljs.io/docs/en/babel-plugin-transform-react-jsx#fragments

**Comment by buck06191** [comment]
Does this not work with regards to Babel and Fragments: https://babeljs.io/docs/en/babel-plugin-transform-react-jsx#fragments ?

File: babel.config.js [hunk 2]
```diff
@@ -1,10 +1,19 @@
 module.exports = {
-    plugins: ["const-enum", "@babel/transform-typescript"],
+    plugins: [
+        "const-enum",
+        "@babel/transform-typescript",
+        [
+            "babel-plugin-emotion",
+            {
+                "cssPropOptimization": true
```
Comment (SiAdcock) [inline babel.config.js:8]: [This is a default](https://emotion.sh/docs/@emotion/babel-plugin#csspropoptimization), I don't think it needs to be explicit

**Comment by SiAdcock** [comment]
Is it safe to remove the [`emotion`](https://github.com/guardian/discussion-rendering/pull/411/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L70) dependency?

File: src/App.stories.tsx [hunk 3]
```diff
@@ -1,6 +1,7 @@
-import React from 'react';
+/** @jsxRuntime classic */
+/** @jsx jsx */
```
Comment (buck06191) [inline src/App.stories.tsx:2]: We should only need the pragma if we're [not using the `babel-preset-css-prop` plugin](https://emotion.sh/docs/css-prop#jsx-pragma)

Do you know why we need both here?

File: src/App.stories.tsx [hunk 4]
```diff
@@ -1,6 +1,7 @@
-import React from 'react';
+/** @jsxRuntime classic */
```
Comment (SiAdcock) [inline src/App.stories.tsx:1]: I think this is only required from [Babel 7.9.0](https://babeljs.io/docs/en/babel-preset-react/). Do you plan on updating that?

Comment (buck06191) [inline src/App.stories.tsx:1]: Can we use [`@babel/plugin-transform-react-jsx`](https://babeljs.io/docs/en/babel-plugin-transform-react-jsx#usage) to avoid this by setting the run time at a higher level?

File: src/components/Avatar/Avatar.stories.tsx [hunk 5]
```diff
@@ -5,7 +5,7 @@ export default { component: Avatar, title: 'Avatar' };
 
 export const Sizes = () => {
 	return (
-		<>
+		<React.Fragment>
```
Comment (buck06191) [inline src/components/Avatar/Avatar.stories.tsx:8]: As above, can we use[`@babel/plugin-transform-react-jsx
`](https://babeljs.io/docs/en/babel-plugin-transform-react-jsx#usage) to avoid this?

File: babel.config.js [hunk 6]
```diff
@@ -1,10 +1,19 @@
 module.exports = {
-    plugins: ["const-enum", "@babel/transform-typescript"],
+    plugins: [
+        "const-enum",
```
Comment (oliverlloyd) [inline babel.config.js:3]: I don't think we need this anymore? Not really relevant to emotion so it doesn't need to be changed here but worth being aware of.

File: src/App.stories.tsx (hunk 3 above)
Comment (SiAdcock) [inline src/App.stories.tsx:2]: The [Emotion docs imply](https://5faaafd0bd0f3f0008469537--emotion.netlify.app/docs/css-prop#get-started) that if you're using the css prop preset, you shouldn't need to use the JSX pragma as well. Have you found this _not_ to be the case?

File: package.json [hunk 7]
```diff
@@ -46,7 +46,9 @@
     "@babel/preset-env": "^7.5.5",
     "@babel/preset-react": "^7.0.0",
     "@babel/preset-typescript": "^7.12.7",
-    "@emotion/babel-preset-css-prop": "^10.0.14",
+    "@emotion/babel-preset-css-prop": "10.0.23",
+    "@emotion/core": "10.0.35",
+    "@emotion/eslint-plugin": "^11.0.0",
```
Comment (oliverlloyd) [inline package.json:51]: Is this related to the emotion changes?

File: package.json [hunk 8]
```diff
@@ -46,7 +46,9 @@
     "@babel/preset-env": "^7.5.5",
     "@babel/preset-react": "^7.0.0",
     "@babel/preset-typescript": "^7.12.7",
-    "@emotion/babel-preset-css-prop": "^10.0.14",
+    "@emotion/babel-preset-css-prop": "10.0.23",
+    "@emotion/core": "10.0.35",
```
Comment (oliverlloyd) [inline package.json:50]: Adding twice because we _really_ want this peer dependency? 😉

File: .storybook/preview.js [hunk 9]
```diff
@@ -25,7 +25,7 @@ const guardianViewports = {
   phablet: {
       name: 'phablet',
       styles: {
-          width: '660px',
+          width: '660px',
```
Comment (oliverlloyd) [inline .storybook/preview.js:28]: Was it this change that caused the diffs in Chromatic?

File: src/components/CommentContainer/CommentContainer.tsx [hunk 10]
```diff
@@ -140,10 +142,10 @@ export const CommentContainer = ({
 				onRecommend={onRecommend}
 			/>
 
-			<>
+			<React.Fragment>
```
Comment (oliverlloyd) [inline src/components/CommentContainer/CommentContainer.tsx:145]: I'm not following along but is this now how we need to do these fragments now? That would be a shame.

**Review by oliverlloyd — COMMENTED** [review]
Great to see this working - nice work!

File: src/components/Avatar/Avatar.stories.tsx (hunk 5 above)
Comment (SiAdcock) [reply src/components/Avatar/Avatar.stories.tsx:8]: This is part of the [React Babel Preset](https://babeljs.io/docs/en/babel-preset-react), I would expect it to be already available.

Comment (SiAdcock) [reply src/components/Avatar/Avatar.stories.tsx:8]: I suspect that the problem is with the elderly version of `@babel/preset-react` that we're currently using. @liywjl can you try upgrading to v7.9.0 or later?

File: .storybook/preview.js (hunk 9 above)
Comment (liywjl) [reply .storybook/preview.js:28]: I have no idea, it seems to have changed focus?? :/ a little concerning

**Comment by mxdvl** [comment]
Closing this as we migrated to Emotion 11 and `@emotion/react` in #454
